src/Sniffs/FinalControllerSniff.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JumpTwentyFour\PhpCodingStandards\Sniffs;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SlevomatCodingStandard\Helpers\TokenHelper;
+
+class FinalControllerSniff implements Sniff
+{
+    public const CONTROLLER_NOT_FINAL = 'ControllerNotFinal';
+
+    public function register(): array
+    {
+        return [
+            T_CLASS,
+        ];
+    }
+
+    public function process(File $phpcsFile, $classPointer): void
+    {
+        $previousPointer = TokenHelper::findPreviousEffective($phpcsFile, $classPointer - 1);
+
+        $tokens = $phpcsFile->getTokens();
+
+        $className = $phpcsFile->getDeclarationName($classPointer);
+
+        if (str_contains($className, 'Test')) {
+            return;
+        }
+
+        if (str_contains($className, 'Controller') === false) {
```

Comment:
Will this kick off about base Laravel controllers that extend the framework's one?

Reply:
Good point, this should probably exclude the default `Controller.php`.

Reply:
I told Matt his and he sneakily put it in a comment as his idea :D

Reply:
You told me to put it on here 😆

Reply:
@AlexJump24 @mmorganJ24 I added an early return for the rule if the classname is just controller return. But we can also add excludes in the ecs.php config when we need to as well